@re-/assert/test/assert.test.ts

```diff
@@ -419,3 +438,57 @@ secondLine`)
         )
     })
 })
+describe("inline meta tests", () => {
+    //sourceFile is cleanedup before every test when "getTemplateFileData" is called
+    let source: SourceFile
+    let templateText: string
+    after(async () => {
+        await cleanUpSourceFile(source, templateText)
+    })
+    it("Checks snap gets populated - precache: true", async () => {
+        const testData: TestData = await getTestFileData(
+            snapShotMetaFile,
+            snapshotTemplate
+        )
+        source = testData.sourceFile
+        templateText = testData.initialText
+        const errors = checkIfTestsContainErrors(
+            testData.statements,
+            testData.expected,
+            testData.fullText
+        )
+        strict.deepEqual(errors, false)
+    }).timeout(9999)
+
+    it("Checks snap gets populated - precache: false", async () => {
+        const testData = await getTestFileData(
+            snapShotMetaFile,
+            snapshotTemplate,
+            false
+        )
+        const errors = checkIfTestsContainErrors(
+            testData.statements,
+            testData.expected,
+            testData.fullText
+        )
+        strict.deepEqual(errors, false)
+    }).timeout(9999)
+})
+describe("bench", () => {
+    let source: SourceFile
+    let templateText: string
+    after(async () => {
+        await cleanUpSourceFile(source, templateText)
+    })
+    it("checks that bench set some kind of value", async () => {
+        const testData: TestData = await getTestFileData(
+            benchMetaFile,
+            benchTemplate,
+            false
+        )
+        source = testData.sourceFile
+        templateText = testData.initialText
+        const errors = checkThatBenchSnapGetsPopulated(testData.statements)
+        strict.deepEqual(errors, false)
```

Comment:
Instead of returning whether there were errors here, we should just throw from the original function if something doesn't validate. That will in turn fail the test and will give a clearer explanation of why it failed, whereas this would just say `true !== false` which would require further investigation.